### PR TITLE
feat: json support

### LINKED ISSUE #12: Json support
- `json` feature flag/optional `serde`/`serde_json` deps
- `.json<T>()` method for `Message`.
- `Json(serde::Error)` case for `Error`.

Just let me know if the feature request gets approved, and I'll go ahead with the pull request.

### PR DESCRIPTION
**Pull request by martsokha**
Closes #12

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "reqwest-websocket"
-version = "0.3.0"
+version = "0.3.1"
```
Comment (jgraef) [inline Cargo.toml:3]: I use [`cargo release`](https://github.com/crate-ci/cargo-release) which will bump this version number when I release the next version.

File: changelog.md [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+  - add `Message::*_from_json` and `Method::json` methods.
+  - add `Error::Json(serde_json::Error)` variant.
 
 # 0.3.0
 
- - close: (#11)
-   - add `protocol::CloseReason`
-   - add `WebSocket::close` method
-   - move `Message` to `protocol`, but re-exported to root.
+- close: (#11)
```
Comment (jgraef) [inline changelog.md:11]: the close changes will actually land in 0.3.1. my mistake. I can fix this once your PR is merged.

**Comment by jgraef** [comment]
looks good! thank you :blush:

Let me know when you're done and I'll merge it.

File: Cargo.toml (hunk 1 above)
Comment (martsokha) [reply Cargo.toml]: Alright! Reverted version in `Cargo.toml`.

File: changelog.md (hunk 2 above)
Comment (martsokha) [reply changelog.md:13]: Fixed!

**Comment by martsokha** [comment]
I've fixed a few more Clippy complaints and applied the requested changes. 
Unfortunately, the version bump is major due to `Error` being an exhaustive enum and having a new variant. 
Could you please take another look?

**Comment by jgraef** [comment]
thank you so much for all the changes! I will take a closer look at it tomorrow.

**Comment by jgraef** [comment]
I'm making `Error` `#[non_exhaustive]`. But I think the added variant doesn't break semver-compatibility, because it's behind a feature flag that wasn't available before.

**Comment by jgraef** [comment]
I changed some minor things, squashed commits and merged this into main. Thanks for this big contribution!

**Pull request closed without merging**